src/editors/containers/ProblemEditor/components/EditProblemView/QuestionWidget/index.jsx

```diff
@@ -1,51 +1,44 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Editor } from '@tinymce/tinymce-react';
 import { connect } from 'react-redux';
-import { injectIntl, FormattedMessage } from '@edx/frontend-platform/i18n';
+import { injectIntl, FormattedMessage, intlShape } from '@edx/frontend-platform/i18n';
 import * as hooks from '../../../hooks';
 import { selectors, actions } from '../../../../../data/redux';
 import { messages } from './messages';
 import './index.scss';
 
-import 'tinymce';
-import 'tinymce/themes/silver';
-import 'tinymce/skins/ui/oxide/skin.css';
-import 'tinymce/icons/default';
-import 'tinymce/plugins/link';
-import 'tinymce/plugins/lists';
-import 'tinymce/plugins/table';
-import 'tinymce/plugins/hr';
-import 'tinymce/plugins/codesample';
-import 'tinymce/plugins/emoticons';
-import 'tinymce/plugins/emoticons/js/emojis';
-import 'tinymce/plugins/charmap';
-import 'tinymce/plugins/code';
-import 'tinymce/plugins/autoresize';
-import 'tinymce/plugins/image';
-import 'tinymce/plugins/imagetools';
+import TinyMceWidget from '../../../../../sharedComponents/TinyMceWidget';
 
 export const QuestionWidget = ({
   // redux
   isLibrary,
   question,
   updateQuestion,
+  assets,
+  lmsEndpointUrl,
+  studioEndpointUrl,
+  // injected
+  intl,
 }) => {
-  const { refReady, setEditorRef } = hooks.prepareEditorRef();
+  const { editorRef, refReady, setEditorRef } = hooks.prepareEditorRef();
   if (!refReady) { return null; }
   return (
     <div className="question-widget">
       <div className="h4 mb-3">
         <FormattedMessage {...messages.questionWidgetTitle} />
       </div>
-      <Editor {
-          ...hooks.problemEditorConfig({
-            isLibrary,
-            setEditorRef,
-            question,
-            updateQuestion,
-          })
-        }
+      <TinyMceWidget
+        editorType="problem"
+        editorRef={editorRef}
+        textValue={question}
+        isLibrary={isLibrary}
+        updateQuestion={updateQuestion}
+        setEditorRef={setEditorRef}
+        minHeight={150}
+        placeholder={intl.formatMessage(messages.placeholder)}
+        assets={assets}
```

Comment:
Should assets be grabbed from redux here instead of being passed down throughout problem editor?

Reply:
I am passing down assets because it is also needed in outside of the tinyMCE editor in the text editor. So I pass it down to reduce the calls to the redux store.

Reply:
What's the benefit of that? Just for curiosity. I am not so deep into redux best practices yet.

Reply:
I am not exactly sure, but it feels wrong to make the call to the redux store in both the parent component and child component.

Reply:
> Should assets be grabbed from redux here instead of being passed down throughout problem editor?

I totally misread your question at first. I thought we were asking why I don't grab assets in `TinyMceEditor`. I can definitely change the code to grab assets in `QuestionWidget`.